app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -129,25 +153,91 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
-  const handleCritterSave = async (animal: IAnimal) => {
-    const critter = new Critter(animal);
+  const renderAnimalFormSafe = (): JSX.Element => {
+    const initialValues = obtainAnimalFormInitialvalues(animalFormMode);
+    if (!initialValues) {
+      return (
+        <YesNoDialog
+          dialogTitle={'Error'}
+          dialogText={'Could not obtain existing critter values.'}
+          open={openAddCritterDialog}
+          onClose={toggleDialog}
+          onNo={toggleDialog}
+          onYes={toggleDialog}
+          noButtonLabel="OK"
+          yesButtonProps={{ sx: { display: 'none' } }}></YesNoDialog>
+      );
+    } else {
+      return (
+        <EditDialog
+          dialogTitle={
+            <Box>
+              <HelpButtonTooltip content={SurveyAnimalsI18N.animalIndividualsHelp}>
+                <Typography variant="h3">Individuals</Typography>
+              </HelpButtonTooltip>
+              <Typography component="span" variant="subtitle1" color="textSecondary" mt={2}>
+                {`${
+                  animalCount
+                    ? `${animalCount} ${pluralize(animalCount, 'Animal')} reported in this survey`
+                    : `No individual animals were captured or reported in this survey`
+                }`}
+              </Typography>
+            </Box>
+          }
+          open={openAddCritterDialog}
+          onSave={(values) => {
+            handleCritterSave(values);
+          }}
+          onCancel={toggleDialog}
+          component={{
+            element: (
+              <IndividualAnimalForm
+                critter_id={currentCritterbaseCritterId}
+                mode={animalFormMode}
+                getAnimalCount={setAnimalCount}
+              />
+            ),
+            initialValues: initialValues,
+            validationSchema: AnimalSchema
+          }}
+        />
+      );
+    }
+  };
+
+  const handleCritterSave = async (currentFormValues: IAnimal) => {
     const postCritterPayload = async () => {
+      const critter = new Critter(currentFormValues);
+      toggleDialog();
       await bhApi.survey.createCritterAndAddToSurvey(projectId, surveyId, critter);
       refreshCritters();
-      dialogContext.setSnackbar({
-        open: true,
-        snackbarMessage: (
-          <Typography variant="body2" component="div">
-            {'Animal added to Survey'}
-          </Typography>
-        )
-      });
+      setPopup('Animal added to survey.');
+    };
+    const patchCritterPayload = async () => {
+      const initialFormValues = obtainAnimalFormInitialvalues(ANIMAL_FORM_MODE.EDIT);
+      if (!initialFormValues) {
+        throw Error('Could not obtain initial form values.');
```

Comment:
Throwing here won't crash the app, but it may provide a worse experience than just showing this text in a snackbar or dialog instead.

Reply:
I do have it pop up a snack bar when catching these, would probably be better to have the specific message placed in the snack bar though.